app/src/main/java/com/internship/move/feature/splash/SplashFragment.kt

```diff
@@ -31,7 +31,19 @@ class SplashFragment : Fragment(R.layout.fragment_splash) {
 
         viewLifecycleOwner.lifecycleScope.launch {
             if (viewModel.getLoggedStatus()) {
-                findNavController().navigate(SplashFragmentDirections.actionGlobalRegisterFragment())
+                if (viewModel.getAuthStatus().isNullOrEmpty() || viewModel.getAuthStatus()
+                        .equals(NULL_USER)
+                ) {
+                    logTag("USERDATA", viewModel.getAuthStatus().toString())
+                    findNavController().navigate(SplashFragmentDirections.actionGlobalRegisterFragment())
+                } else {
+                    logTag("USERDATA", viewModel.getAuthStatus().toString())
+                    val userResponse: UserResponse = Gson().fromJson(
```

Comment:
You don't need Gson since you use moshi.

Also, all this deserialization shouldn't be done in the UI layer, not even in the VM Layer.
You should be receiving the User model ready to use from the other layers.